sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala

```diff
@@ -3537,6 +3537,12 @@ class DataFrameSuite extends QueryTest
       }.isEmpty)
     }
   }
+
+  test("SPARK-41219: Decimal changePrecision should work with decimal(0, 0)") {
+    val df = Seq("0.5944910").toDF("a")
+    checkAnswer(df.selectExpr("cast(a as decimal(7,7)) div 100"), Row(0))
+    checkAnswer(df.select(lit(BigDecimal(0)) as "c").selectExpr("cast(c as decimal(0,0))"), Row(0))
```

Comment:
hmmm, `decimal(0, 0)` is a valid decimal type? how do you use it in production?

Reply:
The numeric equivalent to CHAR(0). The only values are NULL and 0. I would choose the path of least resistance. I agree that there is risk in support. What's the origin of this PR?

Reply:
Now `cast(a as decimal(7,7)) div 100` fails and we want to fix it.

Reply:
Shall we change the evaluation of `decimal div integer` instead?

Reply:
@gengliangwang we can, it actually is the path of least resistance. By making sure the result decimal precision bigger than 0, see the comment https://github.com/apache/spark/pull/38760#discussion_r1030015079.

Besides, some other quries would fail with decimal(0, 0), it can less happen in production though. So this pr wants to make sure how does Spark handle decimal(0, 0) or leave it as is.